### PR TITLE
Skip any attachment fetches that fail with OLE conversion error

### PR DESCRIPTION
**Pull request by meain**
These files from what we can understand are not available and thus can't be fetched. This change ensures that we don't fail the backup just because of this error.

---

#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [x] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* #<issue>

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/pkg/fault/item.go
Comment (ryanfkeepers) [inline src/pkg/fault/item.go:18]: I'm not entirely confident about the choice of "attachment" as a type here, since it's specific to certain types of data and not others.  That sets us up to have N different types, instead of one type with N different identifiers.  For example, if this were, I dunno, an `ItemPropertyType`, with a category of `Attachment`, then we only need one interface for all ItemProperty skips, all of which can dynamically declare different categories.  Easier to extend and maintain that way.  (similary, we should probably change "fileType" to "itemType", one category of which can be a File.)

Not entirely sure what to call it.  Maybe "attachment" is sufficient for the short term?  Don't want to block on this, since we should be able to rename in a fast follow without much fuss.  But do help me think on the best name, please.

File: src/pkg/services/m365/api/mail.go
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/mail.go:415]: This is going to be a problem.  Fault items deduplicate [based on the item id](https://github.com/alcionai/corso/blob/main/src/pkg/fault/item.go#L83).  This skip would be read as "we are skipping the entire item", when you mean for it to indicate that we're skipping just the attachment within that item.  If you can use the attachment ID instead, and add the item ID into the extended map, that would solve the issue.

Also, please coordinate with @juleslasarte about the sdk expectations here.  The introduction of subcomponent skips is a new paradigm, and other frameworks need to handle it gracefully.  As a v1, it might be smarter for us to not include a fault skip at all, and instead log and continue.  That will unblock the fix, and we can investigate the fault skip records as a follow-up.

File: CHANGELOG.md
Comment (ryanfkeepers) [inline CHANGELOG.md:21]: So we'll skip the attachment to progress the backup.  I assume we'll need the item to appear in another delta if we want to retry the attachment download, since we have no other way to target a single item for backup, meaning once skipped, we'll never be able to retrieve the attachment.  That actually sounds like a condition we'd want to list in `Known Issues`.  Have you double checked with Georgi and Vaibhav that we're okay with the permanently missing data here?  Or would it be better to record this situation as a recoverable error?

**Review by ryanfkeepers — APPROVED** [review]
Generally looks good.  Some concerns about using fault skips to handle this issue (at least, as a v1), and want to double check that we're happy with the overall behavior.  Approving to unblock, and you can chase those issues in the background.

File: src/pkg/services/m365/api/mail.go
Comment (juleslasarte) [reply src/pkg/services/m365/api/mail.go:415]: > Also, please coordinate with @juleslasarte about the sdk expectations here. The introduction of subcomponent skips is a new paradigm, and other frameworks need to handle it gracefully. As a v1, it might be smarter for us to not include a fault skip at all, and instead log and continue. That will unblock the fix, and we can investigate the fault skip records as a follow-up.

I second this -- to simply skip for now, and add the fault records as a follow up so we don't need to coordinate on the work on action details and this.

Comment (meain) [reply src/pkg/services/m365/api/mail.go:415]: I've created https://github.com/alcionai/corso/issues/3620 to track this

File: CHANGELOG.md
Comment (meain) [reply CHANGELOG.md:21]: > .. I assume we'll need the item to appear in another delta if we want to retry the attachment download ..
>  Have you double checked with Georgi and Vaibhav that we're okay with the permanently missing data here?

I had a conversation with Vaibhav and we finalized on skipping this for now. We want the backup to go through with this skipped. Similar to malware, it looks like this data will never be accessible from what we could find.

Comment (ryanfkeepers) [reply CHANGELOG.md:21]: > it looks like this data will never be accessible

Hmm, I didn't catch that there's a sense of permanence here, from the code comments.  Maybe you could add a small note about that somewhere in a comment, so it doesn't get confused for a temporary failure?

**Pull request merged**
